build-system/tasks/build-story-localization.js

```diff
@@ -23,7 +23,11 @@ function getLanguageCodeFallbacks(languageCode) {
   if (!languageCode) {
     return [FALLBACK_LANGUAGE_CODE];
   }
-  const matches = languageCode.match(LANGUAGE_CODE_CHUNK_REGEX) || [];
+  const matches = languageCode.match(LANGUAGE_CODE_CHUNK_REGEX);
+  if (!matches) {
```

Comment:
It did cause an issue, but I found a typing-only way to make this change work now (229c71a)

Details if you're curious: `String.match(...)` returns `RegExpMatchArray | null`. In turn, `RegExpMatchArray` is defined as `extends Array<String>`, but between TS 4.8 and 4.9 the definition of `RegExpMatchArray` its fields diverged enough that the resulting type of `languageCode.match(LANGUAGE_CODE_CHUNK_REGEX) || []` changed from `RegExpMatchArray` to `RegExpMatchArray | []`, which means the type of `fallbackLanguageCodeList` below is now `never` instead of `String`

I fixed this by reverting this change and replacing it with an explicit type declaration for `matches` as `string[]` (229c71a)

(TS Playground example: [4.8.4](https://www.typescriptlang.org/play?ts=4.8.4#code/MYewdgzgLgBAtgQysAFgUwjAvDARAgI2FwDpFkUAKXADwE8AvXAShgB82YBtAXQG4AUAPKoMJAE5oAJgFdgaSgJjKYlAA6SAbgEsQMiADUEAGxlpWWAHwwNaHXsMmzAGiUquuAJIA5TwBVcHldmQSA) vs [4.9.5](https://www.typescriptlang.org/play?ts=4.9.5#code/MYewdgzgLgBAtgQysAFgUwjAvDARAgI2FwDpFkUAKXADwE8AvXAShgB82YBtAXQG4AUAPKoMJAE5oAJgFdgaSgJjKYlAA6SAbgEsQMiADUEAGxlpWWAHwwNaHXsMmzAGiUquuAJIA5TwBVcHldmQSA))

Reply:
appreciate the explanation 👍